activerecord/lib/active_record/validations.rb

```diff
@@ -70,6 +70,10 @@ def valid?(context = nil)
 
     alias_method :validate, :valid?
 
+    def run_validations? # :nodoc:
```

Comment:
Maybe move this to the autosave module instead, since it's only used by autosave?
Would `validate_autosave_associations?` be a more descriptive name?